src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala

```diff
@@ -0,0 +1,178 @@
+package com.softwaremill.mqperf
+
+import java.util.concurrent.atomic.{AtomicInteger, AtomicLong}
+import java.util.concurrent.{ExecutorService, Executors}
+
+import akka.dispatch.ExecutionContexts
+import com.codahale.metrics.MetricRegistry
+import com.softwaremill.mqperf.mq.Mq
+import com.softwaremill.mqperf.util.FakeClock
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers, Suite}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+class ReceiverRunnableTest extends FlatSpec with ReceiverTestKit {
+
+  behavior of "ReceiverRunnable"
+
+  it should "wait for idle timeout before stop" in new ReceiverTestCase {
+    receiverShouldBeRunning
+    queueShouldBeEmpty
+
+    sendMsgAndWaitForReceive()
+    receiverShouldBeRunning
+
+    // Almost max idle time
+    addClockTicks(receiverTimeout - 1.nano)
+    receiverShouldBeRunning
+
+    // Exceed the idle timeout
+    addClockTicks(1.nano)
+    receiverShouldBeStopped
+  }
+
+  it should "reset idle time after receiving any message" in new ReceiverTestCase {
+    receiverShouldBeRunning
+    queueShouldBeEmpty
+
+    sendMsgAndWaitForReceive()
+
+    receiverShouldBeRunning
+    // Almost max idle time
+    addClockTicks(receiverTimeout - 1.nano)
+    receiverShouldBeRunning
+
+    sendMsgAndWaitForReceive()
+
+    receiverShouldBeRunning
+    // Almost max idle time
+    addClockTicks(receiverTimeout - 1.nano)
+    receiverShouldBeRunning
+
+    // Exceed the idle timeout
+    addClockTicks(1.nano)
+    receiverShouldBeStopped
+  }
+
+  it should "not timeout before receiving any message" in new ReceiverTestCase {
```

Comment:
Looks like it. Good to know though that ScalaTest can resolve constructor's code as failing/successful test (or can it? ;)